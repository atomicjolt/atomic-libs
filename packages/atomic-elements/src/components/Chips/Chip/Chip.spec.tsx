import { fireEvent, render, screen } from "@testing-library/react";
import { expect, describe, test, vi } from "vitest";
import { Chip } from ".";

describe("Chip", () => {
  test("matches snapshot", () => {
    const result = render(<Chip>Item</Chip>);
    expect(result.asFragment()).toMatchSnapshot();
  });

  test("onRemove", () => {
    const onRemove = vi.fn();
    render(<Chip onRemove={onRemove}>Item</Chip>);
    fireEvent.click(screen.getByRole("button"));
    expect(onRemove).toHaveBeenCalledOnce();
  });

  test("Can be rendered as a link", () => {
    render(<Chip href="/test">Item</Chip>);
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/test");
  });
});
