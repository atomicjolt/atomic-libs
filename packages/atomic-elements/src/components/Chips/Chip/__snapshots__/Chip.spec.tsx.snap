// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chip > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-keTIit ia-dvfO aje-chip"
  >
    <div
      class="sc-dstKZu gTCzyn"
    >
      Item
    </div>
  </div>
</DocumentFragment>
`;
